main.nf

```diff
@@ -140,392 +165,666 @@ process fastqc {
  */
 
 process trim_galore {
-    tag "reads: $reads"
+
+    tag "$name"
 
     module 'bioinfo-tools'
     module 'FastQC'
     module 'cutadapt'
     module 'TrimGalore'
 
-    cpus 3
-    memory '3 GB'
-    time '8h'
+    cpus 2
+    memory { 3.GB * task.attempt }
+    time { 8.h * task.attempt }
+    errorStrategy { task.exitStatus == 143 ? 'retry' : 'terminate' }
+    maxRetries 3
+    maxErrors '-1'
 
-    publishDir "$results_path/trim_galore"
+    publishDir "${params.outdir}/trim_galore", mode: 'copy'
 
     input:
-    file(reads) from trimgalore_input
+    set val(name), file(reads:'*') from trimgalore_input
 
     output:
-    file '*fq.gz' into trimmed_reads, trimmed_reads_miRdeep2
-    file '*trimming_report.txt'
-    
+    file '*.gz' into trimmed_reads_bowtie, trimmed_reads_bowtie2
+    file '*trimming_report.txt' into trimgalore_results
+
     script:
-    /* Trim Galore should automatically detect the small RNA adapter sequence */
-    """
-    trim_galore --gzip --fastqc_args "-q" ${reads}
-    """
+    /* I'm using the latest version of trimgalore downloaded from github. The version on uppmax has a bug when handling smallRNA-seq data. */
+
+    single = reads instanceof Path
+
+    if(single) {
+      """
+      perl /home/chuanw/trimgalore/trim_galore.pl --small_rna --gzip $reads
+      """
+    } else {
+      """
+      perl /home/chuanw/trimgalore/trim_galore.pl --paired --small_rna --gzip $reads
```

Comment:
I think it is due to the accessibility of sequencers: PE sequencing has a
much shorter queue time, or user don't need to pay for an entire SE
flowcell.

On Fri, Sep 9, 2016 at 2:47 PM, Phil Ewels <notifications@github.com> wrote:

> In main.nf
> <https://github.com/SciLifeLab/NGI-smRNAseq/pull/5#discussion_r78173463>:
>
> >      script:
> > -    /* Trim Galore should automatically detect the small RNA adapter sequence */
> > -    """
> > -    trim_galore --gzip --fastqc_args "-q" ${reads}
> > -    """
> > +    /* I'm using the latest version of trimgalore downloaded from github. The version on uppmax has a bug when handling smallRNA-seq data. */
> > +
> > +    single = reads instanceof Path
> > +
> > +    if(single) {
> > +      """
> > +      perl /home/chuanw/trimgalore/trim_galore.pl --small_rna --gzip $reads
> > +      """
> > +    } else {
> > +      """
> > +      perl /home/chuanw/trimgalore/trim_galore.pl --paired --small_rna --gzip $reads
>
> I wonder if we should remove support for paired end reads? Just throw an
> error at the start of the pipeline if the user supplies paired end data..
>
> Does anyone have a reason why we should ever use PE data?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/SciLifeLab/NGI-smRNAseq/pull/5/files/13eb0f8ceaad82a249b320d7d98242153ed83996#r78173463>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ARdKNGq_mwy4H8h3X9FV_hp5RYCCvnE-ks5qoVVvgaJpZM4J5ACd>
> .
>